### PR TITLE
fixes #29

### LINKED ISSUE #29: Race condition when using `async.times`
When I tested the limiter with the code below
````
var redis = require('ioredis'),
  Limiter = require('ratelimiter'),
  async = require('async'),
  db = redis.createClient(),
  limiter;

limiter = new Limiter({
  id: 'hello',
  db: db
});

async.times(10, function(n, next) {
  limiter.get(next);
}, function(err, res) {
  console.log(res);
});
````

It printed the following

````
[ { total: 2500, remaining: 2500, reset: 1485859856 },
  { total: 2500, remaining: 2499, reset: 1485859856 },
  { total: 2500, remaining: 2499, reset: 1485859856 },
  { total: 2500, remaining: 2499, reset: 1485859856 },
  { total: 2500, remaining: 2499, reset: 1485859856 },
  { total: 2500, remaining: 2499, reset: 1485859856 },
  { total: 2500, remaining: 2499, reset: 1485859856 },
  { total: 2500, remaining: 2499, reset: 1485859856 },
  { total: 2500, remaining: 2498, reset: 1485859856 },
  { total: 2500, remaining: 2498, reset: 1485859856 } ]
````

Probably due to n value not getting updated. Sent a `pr` to try fixing this issue. Comments are appreciated.

### PR DESCRIPTION
**Pull request by kp96**
Race condition when used with `async.times`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -104,21 +104,22 @@ Limiter.prototype.get = function (fn) {
     }
 
     db.multi()
-      .set([count, n - 1, 'PX', ex * 1000 - dateNow, 'XX'])
```
Comment (noamshemesh) [inline index.js]: I'm wondering why we didn't change it before 🤔

File: index.js [hunk 2]
```diff
@@ -104,21 +104,22 @@ Limiter.prototype.get = function (fn) {
...
+      .decr(count)
+      .pexpire([count, ex * 1000 - dateNow])
       .pexpire([limit, ex * 1000 - dateNow])
       .pexpire([reset, ex * 1000 - dateNow])
-      .exec(function (err, res) {
+      .exec(function(err, res) {
         if (err) return fn(err);
         if (isFirstReplyNull(res)) return mget();
-        n = n - 1;
+        n = Array.isArray(res[0]) ? ~~res[0][1] : ~~res[0];
```
Comment (noamshemesh) [inline index.js]: Is it possible that the return value would not be an array?

Comment (kp96) [inline index.js]: If it is `redis` module, then `res[0]` is a string value. If it's `ioredis` then it's an array.

**Comment by noamshemesh** [comment]
In general it looks ok, thanks.
Can you please add one or two tests to simulate this problem to make sure we're fixing it

**Comment by kp96** [comment]
sure

**Comment by noamshemesh** [comment]
Hi, I'll merge it next week. I won't be near a computer till then. 
Regarding Travis. I don't want to drop support for node 0.10 if it's a minor issue

**Comment by kp96** [comment]
The tests are failing randomly. (off by 1 sometimes. best guess is round off errors)

**Pull request merged**